packages/loopring_v3/contracts/amm/libamm/AmmPoolToken.sol

```diff
@@ -10,6 +10,9 @@ import "../../lib/SignatureUtil.sol";
 
 
 /// @title AmmPoolToken
+///        This contract do not maintain the following data to save gas:
+///        S.balanceOf[address(this)], and
```

Comment:
i think this balance is only used when people do a forced exit/withdraw when shutdown. So in normal cases it doesn't impact gas costs, so perhaps still useful to have the default behavior where the real balance is returned for tools like etherscan (vs returning uint96(-1) which may be confusing for people/tools).

Reply:
This PR is created when we support layer1 joins and exits. Now it can indeed be dropped.